### ISSUE TITLE
Idea: Add CI for PRs and Pushes to staging

### ISSUE THREAD

**Issue by drvladb** [benawad/dogehouse#1125]
**Is your feature request related to a problem? Please describe.**
Due to the lack of CI for PRs and Pushes to staging, things like missing commas slip though (ie in bcb2875de290b09b51c30fb0cac6ce82d9d41387)

**Describe the solution you'd like**
Add CI to build the project (and possibly deploy it some ware) to see that the PR does in fact what is says it does.

**Describe alternatives you've considered**
Reading over each PR can be time consuming, even more so when your looking for small issues like typos.

**Comment by drvladb** [comment]
It do this if @benawad is fine with it.

**Comment by tejxv** [comment]
yeah we had netlify's deploy previews for quite some time but seems like ben removed it recently
add it back maybe? @benawad

**Comment by drvladb** [comment]
Hmm, there is staging deploy for the staging branch but not for commits.  

![image](https://user-images.githubusercontent.com/33793511/111368365-2af19f00-866c-11eb-873a-fdb0f74d9651.png)

**Comment by tejxv** [comment]
oh wow, is this under Actions tab? i can't seem to find it

**Comment by drvladb** [comment]
Its ether under the actions tab or under the Red X, Green Check, or Yellow Circle under most commits.
![image](https://user-images.githubusercontent.com/33793511/111370603-d3086780-866e-11eb-8d3b-e7829e0b2db5.png)

**Comment by tejxv** [comment]
okay, thank you!

**Comment by benawad** [comment]
I'd like to use CloudFlare pages but I can't figure out how to get the bot to show up on prs 😅

**Comment by drvladb** [comment]
The bot's behavior is quite strange, I set up a test project and it only builds PRs made within the repository (so staging -> prod will work but drvladb/dogehouse:patch-1 -> benawad/dogehouse:staging will not). There doesn't seem to be a github action for this so writing a action to do this on PRs is not an option. We could use Cloudflare Workers, but I'm not sure ben is good with that.

P.S. Test repo is [drvladb/githubpagestest](https://github.com/drvladb/githubpagestest) ([pulls](https://github.com/drvladb/githubpagestest/pulls))

**Comment by benawad** [comment]
I just contacted cloudflare team they are going to take a look, in the mean time if you want to setup workers, I'm cool with that
